spring-context/src/test/java/org/springframework/format/datetime/standard/InstantFormatterTests.java

```diff
@@ -0,0 +1,100 @@
+package org.springframework.format.datetime.standard;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+import java.text.ParseException;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.Random;
+import java.util.stream.Stream;
+
+import static java.time.Instant.MAX;
+import static java.time.Instant.MIN;
+import static java.time.ZoneId.systemDefault;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SuppressWarnings("ConstantConditions")
+class InstantFormatterTests {
+
+	private final InstantFormatter instantFormatter = new InstantFormatter();
+
+	@ParameterizedTest
+	@ArgumentsSource(ISOSerializedInstantProvider.class)
+	void should_parse_an_ISO_formatted_string_representation_of_an_instant(final String input) throws ParseException {
+		final Instant expected = DateTimeFormatter.ISO_INSTANT.parse(input, Instant::from);
```

Comment:
Please remove `final` from local variable and method parameter declarations. We (typically) do not do that in Spring Framework.